app/services/dataworks_mappers/sdr.rb

```diff
@@ -46,420 +56,508 @@ def perform_map
     private
 
     # Not mapping alternative/subtitles at the moment
-    def titles(element = source.description.title)
-      return [] if element.blank?
+    def titles
+      return [] if (title = source.dig('description', 'title', 0, 'value')).blank?
 
-      [
-        { title: Cocina::Models::Builders::TitleBuilder.full_title(element).first }
-      ]
+      [{ title: }]
     end
 
     # Druid is always present, but DOI may not be
     def identifiers
-      [
-        { identifier: source.externalIdentifier, identifier_type: 'DRUID' }
-      ].tap do |ids|
-        ids.push(doi) if doi.present?
-      end
-    end
+      ids = [{ identifier: source['externalIdentifier'], identifier_type: 'DRUID' }]
 
-    # DOI can be stored in two places, so check both
-    def doi
-      # If it's here, it is not in URL form
-      if source.identification.doi.present?
-        return {
-          identifier: source.identification.doi,
-          identifier_type: 'DOI'
-        }
+      if (doi_id = self.class.doi(source:)).present?
+        ids.push({ identifier: doi_id, identifier_type: 'DOI' })
       end
 
-      return if (doi = source.description.identifier.find { |id| id.type == 'DOI' }).nil?
-
-      # If it's here, it is in URL form
-      {
-        identifier: id_from_url(doi.value),
-        identifier_type: 'DOI'
-      }
+      ids
     end
 
-    # Strip the URL prefix from a DOI (or other identifier) and return just the ID
-    def id_from_url(id_url)
-      URI(id_url).path.delete_prefix('/')
+    def all_contributors
+      source.dig('description', 'contributor')&.map { |c| ContributorMapper.new(c) }&.compact || []
```

Comment:
Perhaps:
```
Array(source.dig('description', 'contributor')).map { |c| ContributorMapper.new(c) }
```

Reply:
I went back and forth on this. It's fine to assume that there will always be at least `description.contributor` for any full cocina object, but not handling nulls means you can't construct small "pieces" of an object in tests without erroring.